### ISSUE TITLE
Add support for contextual/environment arguments for script commands

### ISSUE THREAD

**Issue by zenangst** [zenangst/KeyboardCowboy#45]
I think it would increase usability if arguments could be passed to shell scripts using environmental or contextual variables.

Example:

> If the user selects file(s) in Finder, the file(s) paths can be passed as arguments to an inline or path based script.

If we add support for passing of arguments, you could take it one step further and have one command produce output for the succeeding command(s). That could open up for command chains and ability to produce unique command type(s) that can take output.

Example:

> Output from a script command is sent to an `AnnounceCommand` which has different types, either by using `say command` or notification center to render the result of the script execution result.

**Comment by fushugaku** [comment]
it would also be great to get stuff like variables, for example $clipboardText, $activeDisplay and etc.
Maybe it would be great to have in-built text buffers, for example temp files where you could input-output data from and use their contents as variables. Stuff like this can already be done with some scripting and shortcuts, but it's a bit tedious

**Comment by fushugaku** [comment]
Also got an idea, if you can add a blank key in shortut and then use it as a variable
example -
keybinding: fn+w+blank_key
action: take the name of blank key, create a txt file with the name, like a.txt, paste clipboard to it, save file
keybinding: fn+r+blank_key
action: paste from this blank_key txt file

I have a lot of text snippets mapped to keys - it would greatly improve the process of creating new one's

**Comment by zenangst** [comment]
@fushugaku, so you're saying 'blank key' means 'Any Key'? That means the last part of the sequence can be anything you want, right?

**Comment by fushugaku** [comment]
yep, just any key at the end of sequence

**Comment by zenangst** [comment]
I like this idea, added it to my R&D list 🤓

**Comment by fushugaku** [comment]
Wow, that looks really cool, thank you!

**Comment by zenangst** [comment]
@fushugaku here is a pre-release if you want to take it for a spin 😎 

[Keyboard Cowboy 3.25.5.dmg.zip](https://github.com/user-attachments/files/17820581/Keyboard.Cowboy.3.25.5.dmg.zip)

**Comment by zenangst** [comment]
@fushugaku I thought about that myself, that is why I added a button to the left of the recorder button.

![image](https://github.com/user-attachments/assets/1ae225cf-1dd8-4eb5-ba60-2cd9c59dfc74)

I should probably make that more visible to the user 😁

**Comment by fushugaku** [comment]
oh, I installed the wrong version %)
yep, can see it now, will replace my buffer read-write and return with results

**Comment by zenangst** [comment]
Nice, let me know how it goes!

**Comment by fushugaku** [comment]
tried it like this, didn't work for me, looks like the file is not created for some reason
Pasting last letter is okay, but I guess it doesn't work yet in terminal scripts
<img width="1436" alt="image" src="https://github.com/user-attachments/assets/3d47954f-4b41-4475-ae74-d74408d173ef">

**Comment by zenangst** [comment]
I didn’t try this with pass through, perhaps that and the Any key is in conflict. Will need to investigate a bit.

**Comment by zenangst** [comment]
@fushugaku Found the issue, here is a new version.

[Keyboard Cowboy 3.25.5.dmg.zip](https://github.com/user-attachments/files/17851874/Keyboard.Cowboy.3.25.5.dmg.zip)

Basically, I forgot to add these to the terminal environment. As you can see in my screenshot, I tested this with string interpolations with the type command. Totally forgot that there where to different codepaths.

**Comment by fushugaku** [comment]
works as intended, thank you!

**Comment by fushugaku** [comment]
found a bug :)
this scenario doesn't work:
<img width="1140" alt="image" src="https://github.com/user-attachments/assets/8549c38d-54e2-4ff0-9f91-05e95d5c9d3b">
this does:
<img width="1170" alt="image" src="https://github.com/user-attachments/assets/a9000741-82cb-4fad-8e1e-98bb0cf88891">
the only difference is that i have an extra key after first sequence

**Comment by zenangst** [comment]
@fushugaku Are you running the workflow as concurrent or sequential?
To ensure order, it should use `sequential`.

**Comment by zenangst** [comment]
But I am gonna look into why the number of keys in sequence might affect the outcome.

**Comment by zenangst** [comment]
@fushugaku Yeah then there is something wonky with the keybinding and when to trigger the command.
I'll investigate and get back to you. Thanks again for reporting and beta testing ❤️

**Comment by zenangst** [comment]
Hey @fushugaku, I've tried to replicate this and it works for me. Do you have any conflicting shortcuts?

![image](https://github.com/user-attachments/assets/6012327c-833f-4be7-8901-ab6569107be5)

I've also attached a new version of the code. I made some additional changes to the source code, which might have fixed the issue. Let me know if it works for you!

[Keyboard Cowboy 3.25.5.dmg.zip](https://github.com/user-attachments/files/17922628/Keyboard.Cowboy.3.25.5.dmg.zip)

**Comment by fushugaku** [comment]
Hi!
Thank you, tried it, works with concurrent, but for some reason doesn't work with serial
in serial mode it doesn't run cmd+c shortcut, but in concurrent in runs fine %) 
<img width="1290" alt="image" src="https://github.com/user-attachments/assets/9644b13c-4fc2-4cb0-ab66-a17601c5d400">

**Comment by zenangst** [comment]
@fushugaku did you get to try these with any later versions?
